### ISSUE TITLE
Detection module interface and Symbolic Executor

### ISSUE THREAD

**Issue by JoranHonig** [ConsenSys/mythril-classic#679]
## Description
Currently detection modules are ran after analysis, they are modules that provide a function: `analyze(statespace)` that returns a list of issues. 

These modules will be refactored to work based on callbacks, vs a post processing procedure.  Which will allow us to strictly limit mythrils analysis time.

#### Detection Module 
There should be an interface/abstract class defined for the detection modules which exposes the following information
 - Name of the analysis module
 - SWC ID that is detected by the module
 - Instructions that are hooked
 - Description of the module

#### Executor
There should be a new SymExecWrapper automatically discovers these modules and applies their hooks to the laserEVM before initializing the analysis.
This is blocked by the detection module

### Request
Use a separate pr for the detection module and the executor
